src/main/java/com/neroreflex/pizza/Pizza.java

```diff
@@ -91,79 +87,91 @@ public final String getBotID() {
+                // E se lo e' piazza la richiesta che dovra' prendere in carico
+                if (this.tranci.get(command).isLoaded())
+                    this.tranci.get(command).enqueueRequest(new Request(channel, sender, rawParamsString));
+                else
+                    this.sendMessage(channel, sender + " sorry but the requested plugin is not loaded (yet)");
+            } else if (command.compareTo("help") == 0) {// E se non lo e' invia l'errore
+                this.Help(new Request(channel, sender, rawParamsString));
+            } else if (command.compareTo("info") == 0) {// E se non lo e' invia l'errore
+                this.Info(new Request(channel, sender, rawParamsString));
+            } else {
+                this.enqueueMessage(new Message(channel, "I don't know what I have to do at '" + command + "' request :("));
             }
-            else if ((command.compareTo("plugin") == 0) && (!args.isEmpty())) {
-                if (args.elementAt(0).compareTo("list") == 0) {
-                    this.tranci.entrySet().stream().forEach((entry) -> {
+        } else if (message.substring(0, nickLength).compareTo(this.getNick()) == 0) {
+            this.enqueueMessage(new Message(channel, "type !help to get started"));
+        }
+    }
+    
+    public final void Info(Request helpRequest) {
+        // Dividi la richiesta di informazioni in piu' parti
+        Vector<String> args = helpRequest.GetBasicParse();
+        
+        // Non e' stato specificato nulla
+        if (args.isEmpty()) {
+            this.enqueueMessage(new Message(helpRequest.GetChannel(), "I'm PizzaBot a small and modular IRC bot"));
+        } else if (args.get(0).compareTo("source") == 0) {
+            this.enqueueMessage(new Message(helpRequest.GetChannel(), "My source code <3 https://github.com/NeroReflex/Pizza"));
+        } else {
+            this.enqueueMessage(new Message(helpRequest.GetChannel(), "No info about that"));
+        }
+    }
+    
+    public final void Help(Request helpRequest) {
+        // Dividi la richiesta di aiuto in piu' parti
+        Vector<String> args = helpRequest.GetBasicParse();
+        
+        // Non e' stato specificato un plugin
+        if (args.isEmpty()) {
+            this.tranci.entrySet().stream().forEach((entry) -> {
                         String name = entry.getKey();
                         Trancio trancio = entry.getValue();
                         
-                        this.enqueueMessage(new Message(channel, "Plugin \"" + name + "\" Started at: " + trancio.getDate()));
+                        // Ottengo la stringa della guida
+                        String help = trancio.onHelp();
+                        
+                        // Stampo la guida solo se ha senso (la stringa della guida NON E' vuota)
+                        if (!help.isEmpty())
+                            this.enqueueMessage(new Message(helpRequest.GetChannel(), "!" + name + " " + help));
```

Comment:
Per ora OK, ma il si dovrà cambiare in modo che il messaggio con le informazioni di help arrivi in privato all'utente che l'ha richiesto, altrimenti intasa la chat (specie con molti plugin o con help lunghi).